replication-api-impl/src/test/java/com/connexta/replication/api/impl/worker/WorkerTest.java

```diff
@@ -0,0 +1,1186 @@
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.api.impl.worker;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.connexta.replication.api.AdapterException;
+import com.connexta.replication.api.AdapterInterruptedException;
+import com.connexta.replication.api.NodeAdapter;
+import com.connexta.replication.api.NodeAdapterFactory;
+import com.connexta.replication.api.ReplicationException;
+import com.connexta.replication.api.data.CreateRequest;
+import com.connexta.replication.api.data.CreateStorageRequest;
+import com.connexta.replication.api.data.DeleteRequest;
+import com.connexta.replication.api.data.ErrorCode;
+import com.connexta.replication.api.data.Metadata;
+import com.connexta.replication.api.data.MetadataInfo;
+import com.connexta.replication.api.data.NotFoundException;
+import com.connexta.replication.api.data.OperationType;
+import com.connexta.replication.api.data.ReplicationPersistenceException;
+import com.connexta.replication.api.data.Resource;
+import com.connexta.replication.api.data.ResourceInfo;
+import com.connexta.replication.api.data.ResourceRequest;
+import com.connexta.replication.api.data.ResourceResponse;
+import com.connexta.replication.api.data.Site;
+import com.connexta.replication.api.data.SiteType;
+import com.connexta.replication.api.data.Task;
+import com.connexta.replication.api.data.ddf.DdfMetadataInfo;
+import com.connexta.replication.api.impl.NodeAdapters;
+import com.connexta.replication.api.persistence.SiteManager;
+import com.connexta.replication.api.queue.Queue;
+import com.connexta.replication.api.queue.SiteQueue;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.codice.junit.rules.ClearInterruptions;
+import org.junit.Rule;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class WorkerTest {
```

Comment:
❗️ I would want to see tests cases that shows that if the worker is interrupted, it stops its current processing properly. Interruption is part of the contract for a worker so we need to make sure that for one, the task currently being handled is unlocked, completed, or failed and that the interrupted state of the thread is propagated once the loop finishes.